main/integration-tests/__test__/api-tests/workspace-service-catalogs/stop-workspace-service-catalog.test.js

```diff
@@ -0,0 +1,89 @@
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const { runSetup } = require('../../../support/setup');
+const errorCode = require('../../../support/utils/error-code');
+
+const {
+  createWorkspaceTypeAndConfiguration,
+} = require('../../../support/complex/create-workspace-type-and-configuration');
+const {
+  createDefaultServiceCatalogProduct,
+  deleteDefaultServiceCatalogProduct,
+} = require('../../../support/complex/default-integration-test-product');
+
+describe('Stop workspace-service-catalog scenarios', () => {
+  let setup;
+  let adminSession;
+  let productInfo;
+
+  beforeAll(async () => {
+    setup = await runSetup();
+    adminSession = await setup.defaultAdminSession();
+    productInfo = await createDefaultServiceCatalogProduct(setup);
+  });
+
+  afterAll(async () => {
+    await deleteDefaultServiceCatalogProduct(setup, productInfo);
+    await setup.cleanup();
+  });
+
+  describe('Stop workspace-service-catalog', () => {
+    it('should fail if user is inactive', async () => {
+      const adminSession2 = await setup.createAdminSession();
+      const workspaceName = setup.gen.string({ prefix: 'workspace-service-catalog-test' });
+      const { workspaceTypeId, configurationId } = await createWorkspaceTypeAndConfiguration(
+        productInfo,
+        adminSession,
+        setup,
+      );
+      await adminSession.resources.users.deactivateUser(adminSession2.user);
+
+      const response = await adminSession.resources.workspaceServiceCatalogs.create({
+        name: workspaceName,
+        envTypeId: workspaceTypeId,
+        envTypeConfigId: configurationId,
+      });
+
+      await expect(
+        adminSession2.resources.workspaceServiceCatalogs.workspaceServiceCatalog(response.id).stop(),
+      ).rejects.toMatchObject({
+        code: errorCode.http.code.unauthorized,
+      });
+    });
+
+    it('should fail if user is anonymous', async () => {
+      const anonymousSession = await setup.createAnonymousSession();
+      const workspaceName = setup.gen.string({ prefix: 'workspace-service-catalog-test' });
+      const { workspaceTypeId, configurationId } = await createWorkspaceTypeAndConfiguration(
+        productInfo,
+        adminSession,
+        setup,
+      );
+
+      const response = await adminSession.resources.workspaceServiceCatalogs.create({
+        name: workspaceName,
+        envTypeId: workspaceTypeId,
+        envTypeConfigId: configurationId,
+      });
+
+      await expect(
+        anonymousSession.resources.workspaceServiceCatalogs.workspaceServiceCatalog(response.id).stop(),
+      ).rejects.toMatchObject({
+        code: errorCode.http.code.badImplementation,
+      });
+    });
+  });
```

Comment:
Add should fail if not owner?

Reply:
Yes, I should. Updated this test case and also the "start" test case. Thanks for catching this!